Fix Gazebo resource path issue for GRO830`s lab.

MarcOlivierFecteau opened the pull request:
This commit attempts to fix issues with Gazebo for GRO830's lab by reverting changes made in A24.

NOTES:

- As required when migrating from Ignition to Gazebo, the environment variable for Gazebo's ressource path's prefix is changed to "GZ_SIM".
- (2025-09-18) This commit is awaiting testing: do NOT merge yet!

MarcOlivierFecteau committed:
Ensure required Gazebo resource paths are added to environment variable

This commit aims to fix the final issue with GRO830's lab
(exercise 12.4) by properly setting the GZ_SIM_RESOURCE_PATH environment
variable in the launch file for the Gazebo simulation.

The main culprit for this issue seems to be that the resource paths were
added to the incorrect environment variable. IGN_GAZEBO_RESOURCE_PATH is
for Gazebo Fortress (for ROS2 Humble). From now on, "GZ_SIM" is the
correct prefix to use for environment variables.

NOTES:

- This commit has NOT yet been fully tested on a lab PC, so do NOT merge
yet!

chameau5050 commented on racecar_gazebo/launch/racecar_circuit.launch.py:
```diff
@@ -3,17 +3,18 @@
+    racecar_gazebo = os.path.join(get_package_share_directory('racecar_gazebo'))
 
-    addEnvVariable = AppendEnvironmentVariable("IGN_GAZEBO_RESOURCE_PATH",models_path)
+    gazeboDefaultResourcePath = SetEnvironmentVariable(name="GZ_SIM_RESOURCE_PATH", value="/opt/ros/jazzy/share")
```
I think it would be better to just keep the append because if someone have custom config this would wipe it

MarcOlivierFecteau replied:
Agreed, but we need to test if "Append" works if the environment variable does not exist (i.e. if it creates it) BEFORE merging.

MarcOlivierFecteau replied:
Change applied. Testing required.

chameau5050 reviewed (changes requested):
just a tiny change

MarcOlivierFecteau committed:
Replace `SetEnvironmentVariable` with `AppendEnvironmentVariable`

This should ensure the existing values in one's configuration are not
overwritten by the launch file.

TODO:

- Confirm `AppendEnvironmentVariable` creates the environment variable
if it does not exist, which is the case for the faculty's lab PCs,
BEFORE merging the PR.
